### PR TITLE
Item show page and status

### PR DESCRIPTION
**Pull request by wlffann**
Currently the retired items functionality is working. I added an enum field to items that defaults to active but can be set to retired. 

I'm going to refactor the view and style it. Let me know if you have any ideas or suggestions!

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -4,4 +4,16 @@ def categories
     Category.all
   end
 
+  def item_status_check(item)
```
Comment (kheppenstall) [inline app/helpers/application_helper.rb]: Like the use of the helper here. 👍

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -4,4 +4,16 @@ def categories
...
+    if item.status == "active"
```
Comment (kheppenstall) [inline app/helpers/application_helper.rb]: Since we have the enum could we do `if item.active?` @wlffann

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -4,4 +4,16 @@ def categories
...
+      button_to "Add to Cart", carts_path(item_id: item.id)
+    else
+      button_to "Item Retired", retired_carts_path(item_id: item.id)
+    end
+  end
+
+  def format_price_per_unit(item)
```
Comment (kheppenstall) [inline app/helpers/application_helper.rb]: Thoughts on whether this should go in the helper too. Do we use this in the controller or just the view? @wlffann

File: app/views/shared/_items_index.html.haml [hunk 4]
```diff
@@ -6,17 +6,18 @@
     - category.items.each do |item|
       %section.item-card{:id => item.name}
         %article.item-card-image
-          %img{'src' => item.image_url, 'alt' => item.name}
+          %a{'href' => item_path(item)}
```
Comment (kheppenstall) [inline app/views/shared/_items_index.html.haml]: @wlffann Could we use a `link_to` helper method here?

File: db/seeds.rb [hunk 5]
```diff
@@ -23,6 +23,14 @@
   image: Rack::Test::UploadedFile.new(File.join(Rails.root, 'db', 'images', 'pistachios.png'))
 )
 
+nuts.items.create(
+  name: 'Brazil Nuts',
+  description: 'A creamy texture and sweet taste.',
```
Comment (kheppenstall) [inline db/seeds.rb]: 👍 Lovely description.

**Comment by akintner** [comment]
This all looks good to me. I like the hover styling for buttons.

**Comment by akintner** [comment]
You do now have some merge conflicts that have to be resolved because of Kyle's most recent merge.

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (kheppenstall) [inline app/helpers/application_helper.rb]: I've never used the helpers directory before. Thoughts on whether we should make an `items_helper.rb` or just leave it in `application_helper`?

File: app/helpers/application_helper.rb (hunk 2 above)
Comment (wlffann) [inline app/helpers/application_helper.rb]: We can! You're totally right

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (wlffann) [inline app/helpers/application_helper.rb]: I can create a PORO if you want! This is the only method we need now but it could always take more

**Comment by kheppenstall** [comment]
I left some comments but ready to merge once @wlffann takes a look at them.

**Comment by tmikeschu** [comment]
@wlffann looks great. My question is, what makes an item retired? Would the store manage that based on availability? If so, would we need to incorporate a stock quantity of the item at some point?

**Pull request merged**